app/controllers/application_controller.rb

```diff
@@ -123,11 +123,23 @@ def respond_with_error(code)
   end
 
   def render_cached_json(cache_key, **options)
+    options[:expires_in] ||= 3.minutes
+    cache_key              = cache_key.join(':') if cache_key.is_a?(Enumerable)
+    content_type           = options.delete(:content_type) || 'application/json'
+
     data = Rails.cache.fetch(cache_key, { raw: true }.merge(options)) do
       yield.to_json
     end
 
     expires_in options[:expires_in], public: true
```

Comment:
Sorry I didn't catch that earlier, but you're unconditionally marking it as public here, which may cause problems.
Indeed, I believe it is possible to get private toots rendered with “public: true” using this code. This shouldn't immediately cause issues as “Set-Cookies” headers would be set, preventing the response to be cached, but it's still worrying.